src/services/types.ts

```diff
@@ -1463,6 +1468,9 @@ namespace ts {
 
     /** @internal */
     export interface Refactor {
+        /** list of actions a refactor can provide */
+        actions?: RefactorActionInfo[];

```

Comment:
Yes, I had it that way before and was flip-flopping because I thought there might be some clever reason to have actual actions in there.

Really we can even do the check at the beginning of `getAvailableActions` with little difference, but having them up front makes debugging slightly easier for me since the refactor function never gets called.